esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -2213,3 +2213,21 @@ extern TPWRAPPER_API void initContainerRoxieTargets(MapStringToMyClass<ISmartSoc
     }
 }
 
+extern TPWRAPPER_API unsigned getContainerThorClusterNames(StringArray& targetNames, StringArray& queueNames)
+{
+    Owned<IStringIterator> targets = getContainerTargetClusters("thor", nullptr);
+    ForEach(*targets)
+    {
+        SCMStringBuffer target;
+        targets->str(target);
+        targetNames.append(target.str());
+        queueNames.append(target.s.append(THOR_QUEUE_EXT));
+    }
+    return targetNames.ordinality();
```

Comment:
would it be better if this was implemented by using getContainerTargetClusters (with a type filter) ?
It then provides a method to get the queue name, rather than reconstructing it here.

Reply:
@jakesmith do you mean that I should implement the getContainerWUClusterInfo(with a type filter) ? The getContainerTargetClusters() returns an array of target names.